.github/workflows/aml-nightly.yml

```diff
@@ -0,0 +1,98 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# ---------------------------------------------------------
+
+name: aml-nightly
+
+on:
+  #          ┌───────────── minute (0 - 59)
+  #          │ ┌───────────── hour (0 - 23)
+  #          │ │ ┌───────────── day of the month (1 - 31)
+  #          │ │ │ ┌───────────── month (1 - 12 or JAN-DEC)
+  #          │ │ │ │ ┌───────────── day of the week (0 - 6 or SUN-SAT)                                  
+  #          │ │ │ │ │
+  #          │ │ │ │ │
+  schedule: 
+    - cron: '0 0 * * *'  # basically running everyday at 12AM
+    # cron works with default branch (main) only: # https://github.community/t/on-schedule-per-branch/17525/2
+  
+  push:
+    # because we can't schedule runs for non-main branches,
+    # to ensure we are running the build on the staging branch, we can add push policy for it
+    branches: [staging]
+
+  # enable manual trigger
+  workflow_dispatch:
+    input:
+      tags:
+        description: 'Tags to label this manual run (optional)'
+        default: 'Anything to describe this manual run'
+
+env:
+  CLUSTER_NAME: "nightly-cluster"
+  EXP_NAME: "nightly_tests"
+  RG: "pradjoshi-aml-rg"
+  WS: "pradjoshi-aml-ws"
+  TEST_LOGS_PATH: '"test_logs.log"'
+  PYTEST_EXIT_CODE: "pytest_exit_code.log"
+
+jobs:
+  nightly-tests:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ['"python=3.6"']
```

Comment:
I don't understand this very well. We have an initial python 3.6 version, but then below there is 3.8, why is that?

Also FYI, the version we are using in the ADO pipeline is 3.7 https://github.com/microsoft/recommenders/blob/main/tests/ci/azure_pipeline_test/dsvm_nightly_linux_cpu.yml#L35

Reply:
Python 3.8 is for the GitHub machines that we use to trigger AzureML experiments from. I will change AzureML python version to 3.7 for now to match with ADO, will add other versions (3.6, 3.8, 3.9) once the request for additional vCPUs goes through.